crates/llm-remote-http/src/lib.rs

```diff
@@ -102,22 +94,20 @@ impl LlmEngine for RemoteHttpLlmEngine {
             .headers(headers)
             .body(body)
             .send()
-            .await;
-
-        match resp {
-            Ok(resp) => match resp.json::<InferResponseBody>().await {
-                Ok(val) => Ok(wasi_llm::InferencingResult {
-                    text: val.text,
-                    usage: wasi_llm::InferencingUsage {
-                        prompt_token_count: val.usage.prompt_token_count,
-                        generated_token_count: val.usage.generated_token_count,
-                    },
-                }),
-                Err(err) => Err(wasi_llm::Error::RuntimeError(format!(
-                    "Failed to deserialize response: {err}"
-                ))),
-            },
-            Err(err) => Err(wasi_llm::Error::RuntimeError(err.to_string())),
+            .await
+            .map_err(|err| wasi_llm::Error::RuntimeError(err.to_string()))?;
```

Comment:
Still missing the context from this error. For example if the error is "HTTP 500" the user will have no idea what call resulted in the 500. We should do the same kind of context adding as we do with the deserialization response below.

Reply:
fixed it.